test/Microsoft.Azure.Devices.Edge.Test/PriorityQueues.cs

```diff
@@ -124,32 +128,159 @@ public async Task PriorityQueueModuleToModuleMessages()
+                           ("logAnalyticsSharedKey", "Unnecessary"),
+                           ("logAnalyticsLogType", "Unnecessary"),
+                           ("testStartDelay", "00:00:00"),
+                           ("testDuration", "00:20:00"),
+                           ("verificationDelay", "00:00:00"),
+                           ("STORAGE_ACCOUNT_CONNECTION_STRING", "Unnecessary"),
+                           ("NetworkControllerRunProfile", "Online"),
+                           ("TEST_INFO", "key=unnecessary")
+                       })
+                       .WithSettings(new[] { ("createOptions", "{\"HostConfig\": {\"PortBindings\": {\"5001/tcp\": [{\"HostPort\": \"5001\"}]}}}") })
+
+                       .WithDesiredProperties(new Dictionary<string, object>
+                       {
+                           ["reportMetadataList"] = new Dictionary<string, object>
+                           {
+                               ["reportMetadata1"] = new Dictionary<string, object>
+                               {
+                                   ["TestReportType"] = "CountingReport",
+                                   ["TestOperationResultType"] = "Messages",
+                                   ["ExpectedSource"] = $"{loadGenModuleName}.send",
+                                   ["ActualSource"] = $"{relayerModuleName}.receive",
+                                   ["TestDescription"] = "unnecessary"
+                               }
+                           }
+                       });
+
+                    builder.AddModule(loadGenModuleName, loadGenImage)
+                        .WithEnvironment(new[]
+                        {
+                            ("testResultCoordinatorUrl", trcUrl),
+                            ("senderType", "PriorityMessageSender"),
+                            ("trackingId", trackingId),
+                            ("testDuration", loadGenTestDuration),
+                            ("messageFrequency", "00:00:00.5"),
+                            ("priorities", priorityString),
+                            ("ttls", ttlString),
+                            ("ttlThresholdSecs", ttlThresholdSecs.ToString())
+                        });
+
+                    Dictionary<string, object> routes = this.BuildRoutes(priorityString.Split(';'), ttlString.Split(";"), loadGenModuleName, relayerModuleName);
+                    builder.GetModule(ModuleName.EdgeHub).WithDesiredProperties(new Dictionary<string, object> { ["routes"] = routes });
+                });
+
+            EdgeDeployment deployment = await this.runtime.DeployConfigurationAsync(addInitialConfig, token);
+            PriorityQueueTestStatus loadGenTestStatus = await this.PollUntilFinishedAsync(loadGenModuleName, token);
+
+            // Wait long enough for TTL to expire for some of the messages
+            Log.Information("Waiting for 20 seconds for TTL's to expire");
+            await Task.Delay(20000);
```

Comment:
should it be better to use ttlThresholdSecs?

Reply:
I am worry that if relayer module takes long time to download and start up; messages may be dropped more than expected.  You may need to re-think about how to pick TTLs.

Reply:
Used Richard's suggestion above. This could still be an issue, but much lesser chance now. And can always bump lowest TTLOffset up even higher